common/types_spec_test.go

```diff
@@ -0,0 +1,60 @@
+package common
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSubmitBuilderBlockJSON(t *testing.T) {
+	jsonBytes := LoadGzippedBytes(t, "../testdata/submitBlockPayloadCapella_Goerli.json.gz")
+
+	submitBlockData := new(VersionedSubmitBlockRequest)
+	err := json.Unmarshal(jsonBytes, &submitBlockData)
+	require.NoError(t, err)
+
+	marshalledJSONBytes, err := json.Marshal(submitBlockData)
+	require.NoError(t, err)
+	buffer := new(bytes.Buffer)
+	err = json.Compact(buffer, jsonBytes)
+	require.NoError(t, err)
+	expectedJSONBytes := buffer.Bytes()
+
+	require.Equal(t, expectedJSONBytes, bytes.ToLower(marshalledJSONBytes))
```

Comment:
I see that `bytes.ToLower` is necessary because the proposer fee recipient is mixed-case (checksummed).